modules/common/outputs.tf

```diff
@@ -18,9 +18,8 @@ output "eks" {
   description = "EKS Module"
 }
 
-
 output "efs" {
-  value       = module.efs
+  value       = local.create_share_home && length(module.efs) == 1 ? module.efs : null
```

Comment:
Do we need to check `length(module.efs) == 1` while already checking `local.create_share_home`?

Reply:
Actually is a good question. I have slightly changed this part and return `module.efs[0]` as `efs`, because now we have involved `count` in the module. In this case yes, we need to check the length to make sure there is no break in `plan` stage, when still the object is not created but terraform tried to reference it.